### DISCUSSION TITLE
Feature request: Matrix . Vector dot product

### DISCUSSION THREAD

**Discussion by epizut** [mathnet/mathnet-numerics#827] [Ideas]
Hi,

I am not sure if you are open to feature request but we can track here people interested into a Matrix.DotProduct(Vector) method.

I know we can iterate over rows to apply a row.DotProduct(vector), but I like to avoid looping myself to benefit from parallel and SIMD algos.

Here is a stackoverflow question on the same subject: https://stackoverflow.com/questions/46595010/c-sharp-matrixvector-dotproduct-using-mathnet-library

**Comment by epizut** [comment]
Of course wrapping the vector into a (1, n) matrix will do the trick (is OfArray())

**Reply by IoanThomas06** [reply]
Would you be able to explain/show how wrapping the vector could be used to obtain a dot product?

**Comment by epizut** [comment]
Sorry, I will close this request, it was a misunderstanding of how Matrix.DotProduct() works.
